zanata-war/src/main/java/org/zanata/service/impl/TranslationFileServiceImpl.java

```diff
@@ -117,19 +133,23 @@
     @Override
     public TranslationsResource parseTranslationFile(InputStream fileContents,
             String fileName, String localeId, String projectSlug,
-            String iterationSlug, String docId) throws ZanataServiceException {
-        if (fileName.endsWith(".po")) {
-            return parsePoFile(fileContents, projectSlug, iterationSlug, docId);
-        } else if (hasAdapterFor(fileName)) {
+            String iterationSlug, String docId, Optional<String> documentType) throws ZanataServiceException {
+
+        HProjectIteration version =
+            projectIterationDAO.getBySlug(projectSlug, iterationSlug);
+
+        if(version.getProjectType() == ProjectType.File) {
             File tempFile = persistToTempFile(fileContents);
             TranslationsResource transRes =
-                    parseAdapterTranslationFile(tempFile, projectSlug,
-                            iterationSlug, docId, localeId, fileName);
+                parseAdapterTranslationFile(tempFile, projectSlug,
+                    iterationSlug, docId, localeId, fileName, documentType);
             removeTempFile(tempFile);
             return transRes;
+        } else if(fileName.endsWith(".po")) {
```

Comment:
- [ ] You keep putting if statements without a space (`if(foo` instead of `if (foo`). I think our style has consistently been with a space. Does this pass checkstyle?